Code Coverage Improvement from 33% to 50%

denduluriramasaimohanakhil opened the pull request:
Code Coverage Improvement from 33% to 50% code coverage

Mohanakhil committed:
Code Coverage Improvement from 33% to 50%

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/controller/FormDataControllerTest.java:
```diff
@@ -34,25 +35,22 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.times;
```
What changed in this file?

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledExceptionTest.java:
```diff
@@ -1,463 +1,453 @@
+import static org.mockito.Mockito.when;
+
+@ActiveProfiles(TEST_ACTIVE_PROFILE)
+@SpringBootTest
```
Lets avoid using springboot test

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledExceptionTest.java:
```diff
@@ -1,463 +1,453 @@
+        testFormMetaData.put(FORM_VERSION,1);
+        FormDataSchema formDataSchemaTest=new FormDataSchema(TEST_ID,null,TEST_VERSION,testFormData,testFormMetaData);
+        doReturn(userList).when(mockUserDetails).getUserDetails();
+        Assertions.assertThrows(InvalidInputException.class, () ->
```
There is no positive test case?

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/FormDataAuditServiceImplTest.java:
```diff
@@ -31,8 +30,10 @@
 import org.springframework.test.context.ActiveProfiles;
 import org.springframework.test.context.junit.jupiter.SpringExtension;
 import org.springframework.web.reactive.function.client.WebClient;
+
```
what changed in this file?

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java:
```diff
@@ -131,198 +127,207 @@ public void init()
+            Mockito.when(mockValidationCheckService.allFieldsValidations(any(),any(),any(),any())).thenReturn(list1);
+            mockFormDataServiceImpl.saveFormData(formDataSchemaTest);
+            verify(mockMongoTemplate,times(1)).save(any(),any());
+            verify(mockFormDataAuditServiceImpl,times(1)).saveFormDataAudit(formDataAuditSchemaTest);
```
Good!!

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java:
```diff
@@ -131,198 +127,207 @@ public void init()
+            Mockito.when(mockFormDataAuditServiceImpl.saveFormDataAudit(formDataAuditSchemaTest)).thenReturn(new FormDataAuditResponse(TEST_ID_VALUE, TEST_VERSION));
+            Mockito.when(mockValidationCheckService.allFieldsValidations(any(), any(), any(), any())).thenReturn(list1);
+            mockFormDataServiceImpl.saveFormData(formDataSchemaTest1);
+            Assertions.assertThrows(FormIdNotFoundException.class,()->mockFormDataServiceImpl.saveFormData(formDataSchemaTest));
```
lets not write two test cases as part of 1.please separate them out.

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java:
```diff
@@ -131,198 +127,207 @@ public void init()
+            Mockito.when(mockValidationCheckService.allFieldsValidations(any(), any(), any(), any())).thenReturn(list1);
+            mockFormDataServiceImpl.saveFormData(formDataSchemaTest1);
+            Assertions.assertThrows(FormIdNotFoundException.class,()->mockFormDataServiceImpl.saveFormData(formDataSchemaTest));
+            mockFormDataServiceImpl.saveFormData(formDataSchemaTest1);
```
dont repeat the calls like this

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java:
```diff
@@ -131,198 +127,207 @@ public void init()
+            Mockito.when(mockFormDataAuditServiceImpl.saveFormDataAudit(formDataAuditSchemaTest)).thenReturn(new FormDataAuditResponse(TEST_ID_VALUE,TEST_VERSION));
+            Mockito.when(mockValidationCheckService.allFieldsValidations(any(),any(),any(),any())).thenReturn(list1);
+            Mockito.when(mockWebClientWrapper.webclientRequest(any(),any(),any(),any())).thenReturn(STRING).thenReturn(STRING).thenReturn("1").thenReturn(String.valueOf(new Exception())).thenReturn(String.valueOf(new Exception()));
+            mockFormDataServiceImpl.saveFormData(formDataSchemaTest1);
```
again multiple calls, should not be done.

VaibhavTechsophy replied:
This is not resolved, still there are two calls.

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/FormServiceExceptionTest.java:
```diff
@@ -18,7 +18,6 @@
 import org.springframework.test.context.ActiveProfiles;
 import java.math.BigInteger;
 import java.util.*;
-
```
nothing changed, please rollback

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/ValidationCheckServiceImplTest.java:
```diff
@@ -1,302 +1,294 @@
+        fieldsMap.put(UNIQUE,false);
+        schemaMap.put(NAME,fieldsMap);
+        List<String> response = mockValidationCheckServiceImpl.allFieldsValidations(schemaMap,dataMapTest,TEST_FORM_ID,TEST_ID);
+        assertThat(response).isInstanceOf(List.class);
```
what is being tested? nothing, Please write actual test case.

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/ValidationCheckServiceImplTest.java:
```diff
@@ -1,302 +1,294 @@
+        Mockito.when(iterator.hasNext()).thenReturn(true).thenReturn(false);
+        Mockito.when(iterator.next()).thenReturn(NAME);
+        List<String> response = mockValidationCheckServiceImpl.allFieldsValidations(schemaMap,dataMapTest,TEST_FORM_ID,TEST_ID);
+        assertThat(response).isInstanceOf(List.class);
```
Please validate response properly.

VaibhavTechsophy commented on src/test/java/com/techsophy/tsf/runtime/form/service/ValidationCheckServiceImplTest.java:
```diff
@@ -1,302 +1,294 @@
+        Mockito.when(iterator.hasNext()).thenReturn(false);
+        Mockito.when(iterator.next()).thenReturn(NAME);
+        List<String> response = mockValidationCheckServiceImpl.allFieldsValidations(schemaMap,dataMapTest,TEST_FORM_ID,TEST_ID);
+        assertThat(response).isInstanceOf(List.class);
```
Please validate response properly. and do same for rest of the cases.

VaibhavTechsophy reviewed (changes requested):
Comments added.
Please also add the lint count from dev/branch to the merge request.

Mohanakhil committed:
Testcases

dev:
 coverage:33%
 lint:175
TestCases:
 coverage:54%
 lint:175

 Removed @SpringbootTest annotation and instead started using MockitoExtension

VaibhavTechsophy reviewed (commented):
I am not seeing any updates made to the comments I gave.

Mohanakhil committed:
Testcases

dev:
 coverage:33%
 lint:175
TestCases:
 coverage:53%
 lint:175

 Assertions are corrected

Mohanakhil committed:
Testcases

dev:
 coverage:33%
 lint:175
TestCases:
 coverage:51%
 lint:175

 Assertions are corrected
